pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -97,11 +98,13 @@ async def run(self):
             if self.custom and (self.auto_previous or self.previous_list or self.in_flight):
                 raise ValueError("Specifying previous list for a custom release is not allowed.")
 
-            if not self.skip_get_nightlies:
+            if self.skip_get_nightlies:
+                candidate_nightlies = self.nightlies
+            elif self.ignore_non_x86_nightlies:
+                candidate_nightlies = await self._get_latest_accepted_nightly()
```

Comment:
Looks like setting this option will cause the pipeline completely ignore the `--nightly` parameter. Is there a way to fix that?

Reply:
this is the design, when ignore_non_x86_nightlies set only use latest accepted x86 nightly.
nightly parameter is used with skip_get_nightlies, if skip_get_nightlies is true the input value of nighties set will be used.

Reply:
why not use the x86 nightly if it is provided as an arg?

Reply:
this is what we want, if ignore_non_x86_nightlies is true we only use x86_64 nightly and build the assembly as we did with all arch nightly, we use x86_64 nightly get brew event and get full arch of rhcos from the rhcos version in x86_64 nightly